recipes/mcap/all/conanfile.py

```diff
@@ -0,0 +1,39 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class McapConan(ConanFile):
+    name = "mcap"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/foxglove/mcap"
+    description = "A C++ implementation of the MCAP file format"
+    license = "Apache-2.0"
+    topics = ("mcap", "serialization", "deserialization", "recording")
+
+    settings = ("os", "compiler", "build_type", "arch")
+    requires = ("fmt/8.1.1", "lz4/1.9.3", "zstd/1.5.2")
+    generators = ("cmake", "cmake_find_package")
+
+    _source_root = "source_root"
+    _source_package_path = os.path.join(_source_root, "cpp", "mcap")
```

Comment:
```suggestion
    @property
    def _source_subfolder(self):
        return "source_subfolder"
        
    @property
    def _source_package_path(self):
        return os.path.join(self._source_subfolder, "cpp", "mcap")
```

Try to use common patterns from Conan Center index. It makes easy to review and compare recipes.